utils/make_all.py

```diff
@@ -0,0 +1,170 @@
+
+        rel_path = path.parent.relative_to(folder_path).as_posix()
+
+        # get sub_dict for this folder
+        path_all_dict = all_dict
+        mod_path = path.parent.relative_to(folder_path).parts
+        try:
+            for part in mod_path:
+                path_all_dict = path_all_dict[part]
+        except KeyError:
+            # there is nothing to import
+            continue
+
+        if '*' in skip_children.get(rel_path, []):
+            path_all_dict = {}
+            alls = []
+            auto_content = ['', '# AUTO-GENERATED', '', '__all__ = ()', '']
+        else:
+            path_all_dict = {
+                key: val for key, val in path_all_dict.items() if key not in skip_children.get(rel_path, [])
+            }
+            alls = gather_all(list(mod_path), path_all_dict, skip_children)
+
+            # check for non-unique imports
+            if len(alls + list(path_all_dict)) != len(set(alls + list(path_all_dict))):
+                non_unique[rel_path] = [k for k, v in Counter(alls + list(path_all_dict)).items() if v > 1]
+
+            auto_content = (['', '# AUTO-GENERATED'] +
+                            ['', '# yapf: disable', '# pylint: disable=wildcard-import', ''] +
+                            [f'from .{mod} import *' for mod in sorted(path_all_dict.keys())] + ['', '__all__ = ('] +
+                            [f'    {a!r},' for a in sorted(set(alls))] + [')', '', '# yapf: enable', ''])
+
+        start_content = []
+        end_content = []
+        in_docstring = in_end_content = False
+        in_start_content = True
+        for line in path.read_text(encoding='utf8').splitlines():
+            if not in_start_content and line.startswith('# END AUTO-GENERATED'):
+                in_end_content = True
+            if in_end_content:
+                end_content.append(line)
+                continue
+            # only use initial comments and docstring
+            if not (line.startswith('#') or line.startswith('"""') or in_docstring):
+                in_start_content = False
+                continue
+            if line.startswith('"""'):
+                if (not in_docstring) and not (line.endswith('"""') and not line.strip() == '"""'):
+                    in_docstring = True
+                else:
+                    in_docstring = False
+            if in_start_content and not line.startswith('# pylint'):
+                start_content.append(line)
+
+        new_content = start_content + auto_content + end_content
+
+        path.write_text('\n'.join(new_content).rstrip() + '\n', encoding='utf8')
+
+    return non_unique
+
+
+if __name__ == '__main__':
+    _folder = Path(__file__).parent.parent.joinpath('aiida')
+    _skip = {
+        'cmdline/params': ['arguments', 'options'],
+        'cmdline/utils': ['echo'],
+        'manage': ['tests'],
+        'orm': ['implementation'],
+        'orm/implementation': ['django', 'sqlalchemy', 'sql'],
+        'restapi': ['*'],
```

Comment:
```suggestion
        # skipped since some arguments and options share the same name
        'cmdline/params': ['arguments', 'options'],
        # skipped since the module and its method share the same name
        'cmdline/utils': ['echo'],
        # skipped since this is for testing only not general use
        'manage': ['tests'],
        # skipped since we don't want to expose the implmentation
        'orm': ['implementation'],
        # skipped since both implementations share class/function names
        'orm/implementation': ['django', 'sqlalchemy', 'sql'],
        # skip all since the module requires extra requirements
        'restapi': ['*'],
```